components/apimgt/org.wso2.carbon.apimgt.rest.api.admin.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/admin/v1/impl/KeyManagersApiServiceImpl.java

```diff
@@ -117,13 +84,18 @@ public Response keyManagersKeyManagerIdDelete(String keyManagerId, MessageContex
         APIAdmin apiAdmin = new APIAdminImpl();
         KeyManagerConfigurationDTO keyManagerConfigurationDTO =
                 apiAdmin.getKeyManagerConfigurationById(organization, keyManagerId);
-        apiAdmin.deleteIdentityProvider(organization, keyManagerConfigurationDTO);
-        apiAdmin.deleteKeyManagerConfigurationById(organization, keyManagerConfigurationDTO);
+        if (keyManagerConfigurationDTO != null) {
+            apiAdmin.deleteIdentityProvider(organization, keyManagerConfigurationDTO);
+            apiAdmin.deleteKeyManagerConfigurationById(organization, keyManagerConfigurationDTO);
 
-        APIUtil.logAuditMessage(APIConstants.AuditLogConstants.KEY_MANAGER,
-                new Gson().toJson(keyManagerConfigurationDTO), APIConstants.AuditLogConstants.DELETED,
-                RestApiCommonUtil.getLoggedInUsername());
-        return Response.ok().build();
+            APIUtil.logAuditMessage(APIConstants.AuditLogConstants.KEY_MANAGER,
+                    new Gson().toJson(keyManagerConfigurationDTO), APIConstants.AuditLogConstants.DELETED,
+                    RestApiCommonUtil.getLoggedInUsername());
+            return Response.ok().build();
+        } else {
+            RestApiUtil.handleResourceNotFoundError(RestApiConstants.RESOURCE_KEY_MANAGER, keyManagerId, log);
```

Comment:
Wouldn't it be better if we throw an APIManagementException with an exception code here so that the global thrower will throw it?

Reply:
Since we in RestAPI layer we don't need to throw API Management Exception from here.